bin/warmup

```diff
@@ -0,0 +1,275 @@
+    output_group.add_argument('--output-latex', dest='output_latex', action='store',
+                              type=str, metavar='LATEX_FILENAME', default=None,
+                              help='Output a LaTeX file containing tables.')
+    output_group.add_argument('--output-json', dest='output_json', action='store',
+                              type=str, metavar='JSON_FILENAME', default=None,
+                              help='Output a JSON file containing summary results.')
+    return parser
+
+
+def setup_logging(options):
+    """Setup logging. Logging level passed in on command line."""
+    level_str = options.debug_level.upper()
+    if level_str not in ('DEBUG', 'INFO', 'WARN', 'DEBUG', 'CRITICAL', 'ERROR'):
+        fatal('Bad debug level: %s' % level_str)
+    level = getattr(logging, level_str.upper())
+    logging.root.setLevel(level)
+    stream = logging.StreamHandler()
+    stream.setLevel(level)
+    stream.setFormatter(CONSOLE_FORMATTER)
+    logging.root.handlers = [stream]
+
+
+def check_environment(need_outliers=True, need_changepoints=True, need_latex=True,
+                      need_plots=True):
+    """Check all modules or executables that the user needs will be available."""
+
+    info('Checking environment.')
+    python_path = None
+    pypy_path = None
+    pdflatex_path = None
+    r_path = None
+    python_path = find_executable('python2.7')
+    if python_path is None:
+        fatal('warmup scripts require Python 2.7, and are not likely to work with Python 3.x.')
+    if find_executable('bzip2') is None or find_executable('bunzip2') is None:
+        fatal('Please install bzip2 and bunzip2 to convert CSV files to Krun JSON format.')
+    if need_outliers:
+        pypy_path = find_executable('pypy')
+        if pypy_path is None:
+            warn('You do not appear to have PyPy installed. Some parts of this '
+                 'script may run slowly.')
+    if need_changepoints or need_plots:
+        try:
+            import numpy
+        except ImportError:
+            fatal('Please install the Python numpy library to generate changepoints and / or plots.')
+    if need_changepoints:
+        r_path = find_executable('R')
+        if r_path is None:
+            fatal('Please install R (e.g. r-base) to generate changepoints.')
+        try:
+            import rpy2
+        except ImportError:
+            fatal('Please install and the Python rpy2 library to generate changepoints.')
+    if need_latex:
+        pdflatex_path = find_executable('pdflatex')
+        if pdflatex_path is None:
+            fatal('You do not appear to have pdflatex installed, and so cannot '
+                  'compile LaTeX files. Please install LaTeX (e.g. tetex-live) and try again.')
+    if need_plots:
+        try:
+            import matplotlib
+        except ImportError:
+            fatal('Please install the Python matplotlib library to generate plots.')
+        try:
+            matplotlib.use('Agg')
```

Comment:
Does this need to be in the `ImportError` try block? [I don't know, it just looks a little unusual.]